nornir/plugins/tasks/version_control/gitlab.py

```diff
@@ -19,15 +19,15 @@ def _generate_diff(original: str, fromfile: str, tofile: str, content: str) -> s
     return "\n".join(diff)
 
 
-def _get_repository(session: requests.Session, url: str, repository: str) -> int:
+def _get_repository(session: requests.Session, url: str, repository: str) -> str:
     if "/" in repository:
         pid = repository.replace("/", "%2F")
```

Comment:
Can you use `urllib.parse.quote()` to the URL encoding, instead of just encoding a forward slash manually.
The same will need to happen for file path btw.

Reply:
Sure, i will change it also for the filename..